### DISCUSSION TITLE
How do i use the IAP codes in game???

### DISCUSSION THREAD

**Discussion by Pickledickle69** [thatskymod/Sky-CotL-Scripts#153] [Q&A] [answered]
I really wanna use the IAP codes but I don't know how to, please teach me on how to.

**Comment by thatskymod** [comment, marked as answer]
## About IAP Codes

First, let's understand the purpose of SNC codes. SNC codes are simply an internal identifier for in-app purchase (IAP) items, similar to how stock-keeping units (SKUs) help warehouses track products in their inventory. These codes won't provide free IAPs, despite what the name might suggest.

With that cleared up, let's proceed.

## Method 1: GameGuardian

*Requires Android.*

1. Once in-game, open GameGuardian (click the floating icon).
2. Select the appropriate memory range for your Android version:  
   - Android 9 (API level 28) = **Ca**  
   - Android 10+ (API level 29+) = **Other**
3. Set the result limit high (e.g., 9999).
4. Click the magnifying glass icon (🔍).
5. Search for the UTF-8 string `:SPASSR` (colons included).
6. Batch replace it with the SNC code you want; for instance, `:SNC25` for Ocean Cape.
7. Hide GameGuardian and open the shop.
   - If the item shows a pink question mark, that's not an issue—just proceed and buy it.

**Explanation:** This substitutes the “regular season pass” with a time-limited item, making it visible in your shop.

For general GameGuardian usage, refer to https://gameguardian.net/help/help.html

### Alternative:
- Search for the UTF-8 string `:global_` (colons included).  
- Replace it with: 0.  
- Search again for `:season_`.  
- Replace it with: 0.

**Explanation:** This will reveal **all** time-limited items in your in-app shop.

## Method 2: Canvas

*Requires Android.*

I'm recommending this method since your GitHub account is only three days old. If you're new to modding, this approach may be more suitable.

In the mod, look for options labeled "Unlock IAPs" or "Instant season" (or similar). This should reveal all items without requiring you to manually modify the game's memory values yourself.

## Final Thoughts

These IAP codes are primarily meant for Sky Beta, where the items are essentially free. In Sky Live, SNC codes are mostly phased out as thatgamecompany already unpublished older items from the Play Store. If you somehow manage to access these items regardless, be aware that this puts your account at greater risk of getting banned.

Finally, I recommend checking the [Credits & Resources](https://github.com/thatskymod/Sky-CotL-Scripts/tree/main?tab=readme-ov-file#credits--resources) section, where you'll find helpful resources for learning about modding and connecting with the community.
